File: clients/backend/communities.ts
interface Community {
  id: number;
  platform: number;
  community_id: string;
  discord_log_channel?: string;
  discord_notify_target?: string;
}

export default class Communities {
  static url = `${process.env.BACKEND_URL!}/community`;

  static async create(community_id: string): Promise<Community | undefined> {
    try {
      const response = await fetch(this.url, {
        method: "POST",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
          "User-Agent": `${process.env.npm_package_name}/${process.env.npm_package_version!} node.js/${process.version}`,
          Authorization: `Token ${process.env.BACKEND_API_KEY!}`,
        },
        body: JSON.stringify({ community_id }),
      });
      if (!response.ok)
        throw new Error(
          `POST ${this.url}: ${response.status} ${response.statusText}`,
        );
      return await response.json();
    } catch (error) {
      console.error(error);
    }
  }

  static async read(community_id: string): Promise<Community | undefined> {
    try {
      const response = await fetch(`${this.url}?id=${community_id}`, {
        method: "GET",
        headers: {
          Accept: "application/json",
          "User-Agent": `${process.env.npm_package_name}/${process.env.npm_package_version!} node.js/${process.version}`,
          Authorization: `Token ${process.env.BACKEND_API_KEY!}`,
        },
      });
      if (!response.ok)
        throw new Error(
          `GET ${this.url}?id=${community_id}: ${response.status} ${response.statusText}`,
        );
      return await response.json();
    } catch (error) {
      console.error(error);
    }
  }

  static async update(
    data: Partial<Community>,
  ): Promise<Community | undefined> {
    try {
      const response = await fetch(this.url, {
        method: "PATCH",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
          "User-Agent": `${process.env.npm_package_name}/${process.env.npm_package_version!} node.js/${process.version}`,
          Authorization: `Token ${process.env.BACKEND_API_KEY!}`,
        },
        body: JSON.stringify(data),
      });
      if (!response.ok)
        throw new Error(
          `PUT ${this.url}: ${response.status} ${response.statusText}`,
        );
      return await response.json();
    } catch (error) {
      console.error(error);
    }
  }

  static async delete(community_id: string) {
    try {
      const response = await fetch(`${this.url}?id=${community_id}`, {
        method: "DELETE",
        headers: {
          "User-Agent": `${process.env.npm_package_name}/${process.env.npm_package_version!} node.js/${process.version}`,
          Authorization: `Token ${process.env.BACKEND_API_KEY!}`,
        },
      });
      if (!response.ok)
        throw new Error(
          `DELETE ${this.url}?id=${community_id}: ${response.status} ${response.statusText}`,
        );
    } catch (error) {
      console.error(error);
    }
  }
}

File: clients/backend/communities.py
from os import getenv
from sys import version
from requests import get, post, put, delete


class Communities:
    url = f"{getenv('BACKEND_URL')}/community"

    @classmethod
    def create(cls, community_id: str) -> dict:
        response = post(
            cls.url,
            headers={
                "Accept": "application/json",
                "User-Agent": f"sibyl-discord python/{version}",
                "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
            },
            json={"communityID": community_id},
        )

        response.raise_for_status()
        return response.json()

    @classmethod
    def read(cls, community_id: int) -> dict:
        response = get(
            f"{cls.url}?id={community_id}",
            headers={
                "Accept": "application/json",
                "User-Agent": f"sibyl-discord python/{version}",
                "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
            },
        )

        response.raise_for_status()
        return response.json()

    @classmethod
    def update(cls, data: dict) -> dict:
        response = put(
            cls.url,
            headers={
                "Accept": "application/json",
                "User-Agent": f"sibyl-discord python/{version}",
                "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
            },
            json=data,
        )

        response.raise_for_status()
        return response.json()

    @classmethod
    def delete(cls, community_id: str) -> None:
        response = delete(
            f"{cls.url}?id={community_id}",
            headers={
                "User-Agent": f"sibyl-discord python/{version}",
                "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
            },
        )

        response.raise_for_status()
